### DISCUSSION TITLE
Grain Distribution Fit & Custom Statistical Distribution

### DISCUSSION THREAD

**Discussion by YeohWY** [neperfepx/neper#487] [Q&A] [answered]
Hello,

I having the following equiaxed grain size distribution which I would like to use to generate polycrystals from Neper. I tried fitting it with a log-normal function in MATLAB (and also looked at other functions like Weibull) but they tend to have some contributions at the left tail (see figure below). Alternatively, I thought of using these discrete points directly to create a msfile to generate the polycrystals in Neper directly.

![May2022_BC1_Histogram_LogNFit](https://user-images.githubusercontent.com/64184075/171007755-edbf1401-2235-4dd1-93f6-a7ac46d90104.jpg)

So my questions are:
1. Is there a more appropriate function that can describe such a skewed distribution in grain sizes?
2. How should I define my msfile such that it describes a statistical distribution (instead of being used as exact values) so that I use it to generate polycrystals with a specified domain?
`neper -T -n from_morpho -dim 2 -domain "square(1,1)" -morpho "diameq:custom(filename)"`

Thank you.

**Comment by rquey** [comment, marked as answer]
>     1. Is there a more appropriate function that can describe such a skewed distribution in grain sizes?

I don't think so, but I may be wrong.

>     2. How should I define my msfile such that it describes a statistical distribution (instead of being used as exact values) so that I use it to generate polycrystals with a specified domain?
>        `neper -T -n from_morpho -dim 2 -domain "square(1,1)" -morpho "diameq:custom(filename)"`

It is not an `msfile` (which stands for *multiscale cell file*)... The file format is defined in [Statistical Distributions](https://neper.info/doc/exprskeys.html#statistical-distributions):

> In the case of the custom distribution, the numerical distribution must be provided in the file. The file must contain the x and y values of the distribution on successive lines. The x values must be provided in ascending order and form a regular grid. The distribution must contain at least 3 points and does not need to integrate to 1.

I would approximate the histogram by a smooth distribution.

**Reply by YeohWY** [reply]
Thanks @rquey for your reply - the manual distribution worked. On a similar note, is it also possible to define two sets of aspect ratios for the polycrystals? For example, grains smaller than 100 um have this aspect ratio distribution of A, whereas grains bigger than 100 um have an aspect ratio distribution of B.

**Reply by rquey** [reply]
This would need an aspect-ratio metric (to work in a similar way as `diameq`, `sphericity`, ...), which Neper does not have at the moment, but it would not be difficult to add.
